Modules/_sqlite/blob.c

```diff
@@ -120,7 +120,25 @@ blob_seterror(pysqlite_Blob *self, int rc)
 }
 
 static PyObject *
-inner_read(pysqlite_Blob *self, Py_ssize_t length, Py_ssize_t offset)
+read_single(pysqlite_Blob *self, Py_ssize_t offset)
+{
+    assert(offset <= sqlite3_blob_bytes(self->blob));
```

Comment:
This can be `<`, right? I was looking at this code last night and noticed this. I think inner_read also has a `<=` where it could be `<`.

Reply:
Yeah, good catch. I prefer to handle this in a separate PR / issue. Are you ok with that?

Reply:
...or we could include it here; here's the patch:

<details>
<summary>diff</summary>

```diff
diff --git a/Modules/_sqlite/blob.c b/Modules/_sqlite/blob.c
index d7045f030e..6d28151cc0 100644
--- a/Modules/_sqlite/blob.c
+++ b/Modules/_sqlite/blob.c
@@ -122,8 +122,6 @@ blob_seterror(pysqlite_Blob *self, int rc)
 static PyObject *
 read_single(pysqlite_Blob *self, Py_ssize_t offset)
 {
-    assert(offset <= sqlite3_blob_bytes(self->blob));
-
     unsigned long buf = 0;
     int rc;
     Py_BEGIN_ALLOW_THREADS
@@ -141,7 +139,7 @@ static PyObject *
 read_multiple(pysqlite_Blob *self, Py_ssize_t length, Py_ssize_t offset)
 {
     assert(length <= sqlite3_blob_bytes(self->blob));
-    assert(offset <= sqlite3_blob_bytes(self->blob));
+    assert(offset < sqlite3_blob_bytes(self->blob));
 
     PyObject *buffer = PyBytes_FromStringAndSize(NULL, length);
     if (buffer == NULL) {
@@ -193,6 +191,11 @@ blob_read_impl(pysqlite_Blob *self, int length)
         length = max_read_len;
     }
 
+    assert(length >= 0);
+    if (length == 0) {
+        return PyBytes_FromStringAndSize(NULL, 0);
+    }
+
     PyObject *buffer = read_multiple(self, length, self->offset);
     if (buffer == NULL) {
         return NULL;
```

</details>

Reply:
Either is fine with me, I think it's small enough to just put it in here.

Reply:
All right, I put it in here :)

Reply:
Looks like you didn't push?